packages/rxtranscribe/src/operators/bufferBetweenSilence.js

```diff
@@ -0,0 +1,112 @@
+import {of, iif, generate, EMPTY, forkJoin, BehaviorSubject} from 'rxjs';
+import {finalize, zip, takeLast, share, mergeMap, map, tap, pairwise, reduce, skipWhile, expand, takeWhile, takeUntil, take} from 'rxjs/operators';
+import toVAD, {VadEvents} from './toVAD';
+
+const MIN_CHUNK_SIZE = 10000;
+const VAD_SAMPLE_SIZE = 1000;
+
+/**
+ *
+ * @param vadOptions options that vad uses for the operator
+ * @param _toVAD The VAD operator
+ */
+const bufferBetweenSilence = function bufferBetweenSilence({
+  vadOptions = { sampleRate: 16000 },
+  _toVAD = toVAD,
+}) {
+  return chunk$ => {
+    let offset = 0;
+
+    // Replays the same chunk that was supplied
+    const chunkSub$ = chunk$.pipe(share());
+
+    const pump = (subject$) => ([vadRes, chunk]) => {
+      console.log('Offset', offset);
+      console.log('Chunk Length', chunk.length);
+      // console.log('Pump values', [vadRes, chunk]);
+      offset += VAD_SAMPLE_SIZE;
+      console.log('this should be false', chunk.length - offset < MIN_CHUNK_SIZE);
+      console.log('this should also be false', vadRes === VadEvents.SILENCE);
+      if (chunk.length - offset < MIN_CHUNK_SIZE || vadRes === VadEvents.SILENCE) {
+        console.log('We should not be getting here.');
+        return subject$.complete();
+      }
+      console.log('We should definitely get here.');
+      return subject$.next(offset);
+    };
+
+    const byteOffset$ = new BehaviorSubject(0);
+    const pumpNext = pump(byteOffset$);
+
+    // Creates slices for VAD to run against
+    const slicer$ = byteOffset$.pipe(
+      // tap(r => console.log('Inside the byteOffset', offset)),
+      zip(chunkSub$),
+      // tap(r => console.log('Ran zip on chunk sub', r)),
+      map(([newOffset, chunk]) => (
+        chunk.subarray(
+          chunk.length - newOffset - VAD_SAMPLE_SIZE - 1, chunk.length - newOffset - 1
+        )
+      )),
+      // tap(r => console.log('Got the new offset', r)),
+      _toVAD({...vadOptions}),
+      // tap(r => console.log('Vad Result', r)),
+      map(([,{type}]) => type),
+      zip(chunkSub$),
+      tap(pumpNext),
+      // reduce((acc, val) => val)
+    );
+
+    // const recusivePartioner$ = slicer$.pipe(
+    //   expand(([vadRes, chunk]) => {
+    //     console.log(`Chunk: ${chunk.length}, Offset: ${offset}, VAD: ${vadRes}`)
+    //     if (vadRes === VadEvents.SILENCE || chunk.length - offset < MIN_CHUNK_SIZE) return EMPTY;
+    //     return chunkAndVad$;
+    //   }),
+    // );
+
+    // Set the offset and chunks
+    return slicer$.pipe(
+      // skipWhile(chunk => !isSilentOrTooSmall(chunk)),
+      takeLast(), // Behavior Subject to fix the double emitting should work
```

Comment:
Oh...  I see.  So we're only taking the final result from the slicer...  It takes an entire stream of file chunks and emits a single result from it.

Reply:
Correct. It is there for debugging, but I have not made it to the the slicer operator yet since it does not seem to be iterating properly.